_posts/2020-05-06-#18806.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: pr
+date: 2020-05-06
+title: "Remove is{Empty,Full} flags from CBloomFilter, clarify CVE fix"
+pr: 18806
+authors: [theStack]
+components: ["p2p"]
+host: jnewbery
+status: upcoming
+commit:
+---
+
+## Notes
+
+- A [bloom filter](https://en.wikipedia.org/wiki/Bloom_filter) is a
+  probabilistic data structure. It supports two operations:
+  - _adding_ an element to the filter.
+  - _querying_ an element from the filter.
+
+- If an element has been previously added, then querying for the element will
+  return _true_. If an element has not been added, then querying for the
+  element may return _true_ or _false_. In other words, querying may return a
+  _false positive_, but will never return a _false negitive_.
+
+- See the [wikipedia page](https://github.com/bitcoin/bitcoin/pull/1795) for
+  how a bloom filter is implemented with hash functions onto a bitfield. Note
+  that the false positive rate depends on the size of the filter and the number
+  of hash functions.
+
+- [BIP 37](https://github.com/bitcoin/bips/blob/master/bip-0037.mediawiki)
+  introduced a new method for SPV clients to use bloom filters to track
```

Comment:
could you add another bullet point explaining SPV = simplified payment verification & the idea of light clients vs a full node? its covered in the BIP motivation section, but seems helpful to highlight here. 

or maybe can just quote this section here
> As Bitcoin grows in usage the amount of bandwidth needed to download blocks and transaction broadcasts increases. Clients implementing simplified payment verification do not attempt to fully verify the block chain, instead just checking that block headers connect together correctly and trusting that the transactions in a chain of high difficulty are in fact valid. See the Bitcoin paper for more detail on this mode.

Reply:
Good idea. I've added a link to https://bitcoin.org/en/operating-modes-guide#simplified-payment-verification-spv.